core/include/traccc/clusterization/clustering_config.hpp

```diff
@@ -0,0 +1,32 @@
+/**
+ * traccc library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include <cstdint>
+
+namespace traccc {
+struct clustering_config {
+    unsigned int threads_per_partition;
+    unsigned int max_cells_per_thread;
+    unsigned int target_cells_per_thread;
+    unsigned int backup_size_multiplier;
+
+    constexpr std::size_t max_partition_size() const {
+        return threads_per_partition * max_cells_per_thread;
+    }
+
+    constexpr std::size_t target_partition_size() const {
+        return threads_per_partition * target_cells_per_thread;
+    }
+
+    constexpr std::size_t backup_size() const {
+        return max_partition_size() * backup_size_multiplier;
+    }
```

Comment:
And I just now remembered... Since you call some of these functions in device code, it would be nice to put `TRACCC_HOST_AND_DEVICE` on them. Yes, `constexpr` does make them work, but since CUDA's support for `constexpr` is still an "experimental feature", it would not hurt us to mark these functions explicitly as device functions. :thinking: